src/sbsign/sbsign.c

```diff
@@ -254,6 +254,10 @@ static int verb_sign(int argc, char *argv[], void *userdata) {
         if (dstfd < 0)
                 return log_error_errno(r, "Failed to open temporary file: %m");
 
+        /* Make sure output file has same mode as input file. */
+        if (fchmod(dstfd, st.st_mode & 07777) < 0)
```

Comment:
Wouldn't this mean that when outputting to e.g. fat32 directly we'd potentially bail? I'd ignore the error?